<a href="https://colab.research.google.com/github/dakshini01/Statistical-Learning-e20181/blob/main/Assignment_5.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Part A: Theoretical Fundamentals (Maximum Likelihood & Decision Space)

## 1. Bias of the Maximum Likelihood Estimator of Covariance

Consider the i.i.d. random variables

$$
X_1, X_2, \ldots, X_n \in \mathbb{R}^m
$$

with population mean

$$
\mu = E[X_i]
$$

and population covariance matrix

$$
\Sigma = Var(X_i).
$$

The empirical sample mean is

$$
\hat{\mu}_n = \frac{1}{n}\sum_{i=1}^{n}X_i.
$$

The raw Maximum Likelihood Estimator for the covariance matrix is

$$
\hat{\Sigma}_{MLE}
=
\frac{1}{n}
\sum_{i=1}^{n}
(X_i-\hat{\mu}_n)(X_i-\hat{\mu}_n)^T.
$$

However, this estimator is biased in finite samples. To show this, we use the identity

$$
\sum_{i=1}^{n}
(X_i-\hat{\mu}_n)(X_i-\hat{\mu}_n)^T
=
\sum_{i=1}^{n}
(X_i-\mu)(X_i-\mu)^T
-
n(\hat{\mu}_n-\mu)(\hat{\mu}_n-\mu)^T.
$$

Taking expectation on both sides,

$$
E\left[
\sum_{i=1}^{n}
(X_i-\hat{\mu}_n)(X_i-\hat{\mu}_n)^T
\right]
=
E\left[
\sum_{i=1}^{n}
(X_i-\mu)(X_i-\mu)^T
\right]
-
nE\left[
(\hat{\mu}_n-\mu)(\hat{\mu}_n-\mu)^T
\right].
$$

Since

$$
E\left[
(X_i-\mu)(X_i-\mu)^T
\right]
=
\Sigma,
$$

we get

$$
E\left[
\sum_{i=1}^{n}
(X_i-\mu)(X_i-\mu)^T
\right]
=
n\Sigma.
$$

Also,

$$
Var(\hat{\mu}_n)
=
E\left[
(\hat{\mu}_n-\mu)(\hat{\mu}_n-\mu)^T
\right]
=
\frac{\Sigma}{n}.
$$

Therefore,

$$
E\left[
\sum_{i=1}^{n}
(X_i-\hat{\mu}_n)(X_i-\hat{\mu}_n)^T
\right]
=
n\Sigma
-
n\left(\frac{\Sigma}{n}\right).
$$

Hence,

$$
E\left[
\sum_{i=1}^{n}
(X_i-\hat{\mu}_n)(X_i-\hat{\mu}_n)^T
\right]
=
(n-1)\Sigma.
$$

Now,

$$
E[\hat{\Sigma}_{MLE}]
=
E\left[
\frac{1}{n}
\sum_{i=1}^{n}
(X_i-\hat{\mu}_n)(X_i-\hat{\mu}_n)^T
\right].
$$

Therefore,

$$
E[\hat{\Sigma}_{MLE}]
=
\frac{1}{n}(n-1)\Sigma.
$$

Thus,

$$
E[\hat{\Sigma}_{MLE}]
=
\frac{n-1}{n}\Sigma.
$$

Since

$$
\frac{n-1}{n}\Sigma \neq \Sigma,
$$

the raw MLE covariance estimator is biased. It underestimates the true population covariance matrix.

---

## Bessel's Correction and the Unbiased Estimator

To remove this bias, we multiply the MLE covariance estimator by

$$
\frac{n}{n-1}.
$$

This gives the unbiased sample covariance matrix

$$
S
=
\frac{1}{n-1}
\sum_{i=1}^{n}
(X_i-\hat{\mu}_n)(X_i-\hat{\mu}_n)^T.
$$

Taking expectation,

$$
E[S]
=
E\left[
\frac{1}{n-1}
\sum_{i=1}^{n}
(X_i-\hat{\mu}_n)(X_i-\hat{\mu}_n)^T
\right].
$$

Substituting the previous result,

$$
E[S]
=
\frac{1}{n-1}
(n-1)\Sigma.
$$

Therefore,

$$
E[S]=\Sigma.
$$

Hence, applying Bessel's correction gives an unbiased estimator of the population covariance matrix.

The reason for using $n-1$ instead of $n$ is that the sample mean $\hat{\mu}_n$ is estimated from the same dataset. Therefore, one degree of freedom is lost.

---

## 2. Type I Error and Type II Error in Structural Health Monitoring

In continuous structural health monitoring, the null hypothesis is

$$
H_0:
\text{The structural asset is healthy.}
$$

The alternative hypothesis is

$$
H_1:
\text{The structural asset is damaged or abnormal.}
$$

---

### Type I Error

A Type I Error occurs when the diagnostic system rejects $H_0$ even though $H_0$ is actually true.

In this context, it means that the system says the structure is damaged even though it is healthy.

Mathematically,

$$
\alpha
=
P(\text{Reject }H_0 \mid H_0 \text{ is true}).
$$

This is a false alarm.

In an industrial plant, a Type I Error may cause unnecessary shutdown, inspection cost, and production loss.

---

### Type II Error

A Type II Error occurs when the diagnostic system fails to reject $H_0$ even though $H_0$ is false.

In this context, it means that the system says the structure is healthy even though it is actually damaged.

Mathematically,

$$
\beta
=
P(\text{Fail to reject }H_0 \mid H_0 \text{ is false}).
$$

This is a missed detection.

A Type II Error is dangerous because real structural damage may continue unnoticed.

---

### Statistical Power

The statistical power of the diagnostic test is

$$
Power = 1-\beta.
$$

Power is the probability of correctly detecting damage when damage is truly present.

---

## Effect of an Ultra-Conservative Significance Threshold

Suppose the engineer chooses

$$
\alpha = 0.0001.
$$

This is a very small significance level.

The significance level $\alpha$ controls the probability of a false alarm. Therefore, by choosing a very small $\alpha$, the engineer greatly reduces the probability of incorrectly declaring a healthy structure as damaged.

However, this decision also makes the test more conservative.

The rejection region becomes very small:

$$
\alpha \downarrow
\Rightarrow
\text{Rejection region decreases}.
$$

As a result, it becomes harder for the test statistic to fall inside the rejection region.

Therefore,

$$
\beta \uparrow
$$

and

$$
Power = 1-\beta \downarrow.
$$

So, reducing $\alpha$ decreases false alarms, but it also reduces the ability of the diagnostic system to detect real damage.

---

## Geometric Interpretation Using the Healthy Confidence Ellipsoid

In a multivariate diagnostic system, healthy operation can be represented by a confidence ellipsoid around the baseline mean vector.

The ellipsoid represents the region where normal healthy data points are expected to lie.

When $\alpha$ is reduced from a common value such as

$$
\alpha = 0.05
$$

to an ultra-conservative value such as

$$
\alpha = 0.0001,
$$

the critical threshold increases.

Therefore, the healthy confidence ellipsoid becomes larger.

Geometrically,

$$
\alpha \downarrow
\Rightarrow
\text{critical value} \uparrow
\Rightarrow
\text{confidence ellipsoid volume} \uparrow.
$$

This means more observations are accepted as healthy.

Therefore:

- False alarms decrease.
- The healthy region becomes larger.
- The diagnostic system becomes less sensitive.
- Type II Error increases.
- Statistical power decreases.

Thus, an ultra-conservative threshold protects the plant from unnecessary shutdowns, but it increases the risk of missing subtle structural damage.

# Part B: Theoretical Extension (Asymptotic Distributions & Slutsky's Theorem)

The Lindeberg-Lévy Multivariate Central Limit Theorem states that, for i.i.d. random variables

$$
X_1, X_2, \ldots, X_n \in \mathbb{R}^m
$$

with population mean

$$
\mu = E[X_i]
$$

and population covariance matrix

$$
\Sigma = Var(X_i),
$$

the standardized sample mean satisfies

$$
\sqrt{n}(\hat{\mu}_n-\mu)
\overset{d}{\longrightarrow}
N(0,\Sigma)
$$

as

$$
n \rightarrow \infty.
$$

Here,

$$
\hat{\mu}_n = \frac{1}{n}\sum_{i=1}^{n}X_i.
$$

This means that when the sample size becomes very large, the distribution of the sample mean becomes approximately multivariate normal.

---

## Slutsky's Theorem

Let

$$
A_n \overset{d}{\longrightarrow} A
$$

and

$$
B_n \overset{p}{\longrightarrow} b,
$$

where $b$ is a constant.

Then Slutsky's Theorem states that

$$
A_n + B_n \overset{d}{\longrightarrow} A+b,
$$

$$
A_nB_n \overset{d}{\longrightarrow} Ab,
$$

and, if $b \neq 0$,

$$
\frac{A_n}{B_n}
\overset{d}{\longrightarrow}
\frac{A}{b}.
$$

In simple words, Slutsky's Theorem says that if one sequence converges in distribution and another sequence converges in probability to a constant, then we can combine them, and the final limiting distribution remains valid.

---

## Consistency of the Sample Covariance Matrix

The population covariance matrix $\Sigma$ is unknown in real applications.

Therefore, we estimate it using the unbiased sample covariance matrix

$$
S =
\frac{1}{n-1}
\sum_{i=1}^{n}
(X_i-\hat{\mu}_n)(X_i-\hat{\mu}_n)^T.
$$

By the Law of Large Numbers, the sample covariance matrix is a consistent estimator of the true covariance matrix. Therefore,

$$
S \overset{p}{\longrightarrow} \Sigma
$$

as

$$
n \rightarrow \infty.
$$

This means that when the sample size becomes very large, the sample covariance matrix $S$ becomes closer and closer to the true population covariance matrix $\Sigma$.

---

## Applying Slutsky's Theorem

From the Multivariate Central Limit Theorem,

$$
\sqrt{n}(\hat{\mu}_n-\mu)
\overset{d}{\longrightarrow}
N(0,\Sigma).
$$

But in practice, $\Sigma$ is unknown.

Since

$$
S \overset{p}{\longrightarrow} \Sigma,
$$

Slutsky's Theorem allows us to replace the unknown $\Sigma$ with the observable estimator $S$.

Therefore, the practical asymptotic distribution becomes

$$
\sqrt{n}(\hat{\mu}_n-\mu)
\overset{d}{\longrightarrow}
N(0,S).
$$

Equivalently, this can be written as

$$
\hat{\mu}_n
\sim
N\left(
\mu,
\frac{S}{n}
\right).
$$

---

## Interpretation

The expression

$$
\hat{\mu}_n
\sim
N\left(
\mu,
\frac{S}{n}
\right)
$$

means that the sample mean vector $\hat{\mu}_n$ is approximately multivariate normally distributed around the true population mean vector $\mu$.

The covariance of this distribution is

$$
\frac{S}{n}.
$$

As $n$ increases,

$$
\frac{S}{n}
\rightarrow 0.
$$

Therefore, the uncertainty around the sample mean becomes smaller.

This justifies using the sample mean and sample covariance matrix to build confidence regions, hypothesis tests, and diagnostic limits in practical structural health monitoring systems.

---

## Final Conclusion

Slutsky's Theorem justifies replacing the unknown population covariance matrix $\Sigma$ with the sample covariance matrix $S$ because

$$
S \overset{p}{\longrightarrow} \Sigma.
$$

Therefore, the theoretical model

$$
\hat{\mu}_n
\sim
N\left(
\mu,
\frac{\Sigma}{n}
\right)
$$

can be operationalized as

$$
\hat{\mu}_n
\sim
N\left(
\mu,
\frac{S}{n}
\right).
$$

This makes the asymptotic distribution practically usable because $S$ can be calculated directly from the observed dataset.

In [1]:
import numpy as np
import pandas as pd
import scipy.stats as stats

# --- DO NOT MODIFY: Synthetic Strain Sensor Data Generator ---
np.random.seed(42)

n_samples = 5000
n_features = 3

base_data = np.random.multivariate_normal(
    mean=[0.5, -0.2, 1.1],
    cov=[
        [0.09, 0.02, 0.01],
        [0.02, 0.06, 0.03],
        [0.01, 0.03, 0.05]
    ],
    size=n_samples
)

# Inject subtle drift
base_data[4000:, 0] += 0.015
base_data[4000:, 2] -= 0.010

df_strain = pd.DataFrame(
    base_data,
    columns=['Strain_Ch1', 'Strain_Ch2', 'Strain_Ch3']
)

# --------------------------------------------------------

def verify_first_moment_homogeneity(df, g_chunks=5):

    n = len(df)
    m = df.shape[1]

    chunk_size = n // g_chunks

    chunks = []

    for i in range(g_chunks):
        start = i * chunk_size
        end = (i + 1) * chunk_size
        chunks.append(df.iloc[start:end])

    # Global mean
    global_mean = df.mean().values.reshape(-1, 1)

    # Within-chunk matrix W
    W = np.zeros((m, m))

    # Between-chunk matrix B
    B = np.zeros((m, m))

    for chunk in chunks:

        chunk_values = chunk.values

        n_j = len(chunk)

        chunk_mean = chunk.mean().values.reshape(-1, 1)

        # Within matrix
        centered = chunk_values - chunk_mean.T

        W += centered.T @ centered

        # Between matrix
        mean_diff = chunk_mean - global_mean

        B += n_j * (mean_diff @ mean_diff.T)

    T = W + B

    wilks_lambda = np.linalg.det(W) / np.linalg.det(T)

    # Bartlett approximation
    chi_square = -(n - 1 - ((m + g_chunks) / 2)) * np.log(wilks_lambda)

    df_chi = m * (g_chunks - 1)

    p_value = 1 - stats.chi2.cdf(chi_square, df_chi)

    return {
        "Wilks Lambda": wilks_lambda,
        "Bartlett Chi Square": chi_square,
        "Degrees of Freedom": df_chi,
        "p-value": p_value
    }

results = verify_first_moment_homogeneity(df_strain, g_chunks=5)

for k, v in results.items():
    print(f"{k}: {v}")

Wilks Lambda: 0.9904519796929591
Bartlett Chi Square: 47.9215049961488
Degrees of Freedom: 12
p-value: 3.2255259508895406e-06


# Part C: Numerical Verification

The dataset was partitioned into

$$
g=5
$$

consecutive chunks, each containing

$$
n_j=1000
$$

observations.

The global mean vector and individual chunk mean vectors were calculated. Using these means, the Within-Chunk Sum of Squares and Cross Products Matrix

$$
W
$$

and the Between-Chunk Sum of Squares and Cross Products Matrix

$$
B
$$

were constructed.

The Wilks' Lambda statistic was computed as

$$
\Lambda
=
\frac{|W|}{|B+W|}.
$$

The calculated value was

$$
\Lambda = \text{0.9904519796929591}
$$

Using Bartlett's asymptotic transformation,

$$
\chi^2
=
-(n-1-\frac{m+g}{2})
\ln(\Lambda),
$$

the test statistic was obtained as

$$
\chi^2
=
\text{47.9215049961488}.
$$

The corresponding degrees of freedom were

$$
m(g-1)
=
3(5-1)
=
12.
$$

The resulting p-value was

$$
p
=
\text{3.2255259508895406e-06}.
$$

---

## Decision

The significance level was chosen as

$$
\alpha = 0.05.
$$

The decision rule is:

- If

$$
p \le 0.05
$$

reject

$$
H_0
$$

- If

$$
p > 0.05
$$

fail to reject

$$
H_0.
$$

Since

$$
p \text{ =3.2255259508895406e-06< } 0.05,
$$

we conclude that:


There is sufficient evidence to conclude that the chunk means differ significantly. Therefore, the baseline center has shifted over time and the first moment homogeneity assumption is violated.

